app/controllers/api/v1/packages_controller.rb

```diff
@@ -36,8 +36,12 @@ class PackagesController < Api::V1::ApiController
 
       api :GET, "/v1/packages", "get all packages for the item"
       def index
-        @packages = @packages.find(params[:ids].split(",")) if params[:ids].present?
-        render json: @packages, each_serializer: serializer, include_orders_packages: true
+        records = @packages #security
+        records = records.filter("state" => params["state"]) if params["state"].present?
+        records = records.search({search_text: params["searchText"]}) if params["searchText"]
+        records = records.find(params[:ids].split(",")) if params[:ids].present?
+        records = records.page(params["page"]).per(params["per_page"])
```

Comment:
what if `page` and `per_page` are omitted or are bad? See `page` and `per_page` methods down around line 150 and use those. Perhaps even move those methods to ApiController so that other controllers can use them